### PR TITLE
Tweak AVBitStream related API, Add more tests

### PR DESCRIPTION
**Pull request by ldm0**
Did some tweaking on the AVBitStream API. Could you help reviewing this? @FallingSnow

### CODE REVIEW

File: src/avcodec/bsf.rs [hunk 1]
```diff
@@ -139,28 +188,20 @@ impl Drop for AVBSFContextUninit {
...
     #[test]
-    #[ignore = "get_null returns bsf_list, idk if it's supposed to"]
     fn test_null_filter() {
         let ctx = AVBSFContextUninit::get_null();
-        let filter = unsafe { *ctx.filter };
-        let filter_name = unsafe { CStr::from_ptr(filter.name) };
-
-        let name = std::ffi::CString::new("null").unwrap();
-        assert_eq!(name.as_c_str(), filter_name);
+        assert_eq!(cstr!("bsf_list"), ctx.filter().name());
```
Comment (ldm0) [inline src/avcodec/bsf.rs:205]: https://github.com/FFmpeg/FFmpeg/blob/2fac1e370c9cf48b2e9d4a7f2c0d7236017f9bbe/libavcodec/bsf.c#L556

**Comment by ldm0** [comment]
@FallingSnow Damn, after writing a sample I found that FFmpeg rely on the reusage of packet sending to the `AVBSFContext`. Which means the API you proposed works, but the API on current master doesn't work, which is really evil......

**Comment by FallingSnow** [comment]
Yeah that's strange. I saw that the API likes you to reuse packets but I didn't think it would require it.

File: src/avcodec/bsf.rs (hunk 1 above)
Comment (FallingSnow) [reply src/avcodec/bsf.rs]: Ah! TY

**Comment by ldm0** [comment]
> Yeah that's strange. I saw that the API likes you to reuse packets but I didn't think it would require it.

If you don't reuse the packet, the receive packet will always returns EOF error LOL. I think it may be an implementation bug of `h264_mp4toannexb_bsf`. I'll file a bug report on their mailing list later :-P

**Comment by FallingSnow** [comment]
Code looks good to me, didn't look over the av_splitter test though. I'm gonna use BSF with ffmpeg's `extract_extradata` so I'll let you know if I run into any issues.

**Comment by ldm0** [comment]
> Code looks good to me, didn't look over the av_splitter test though. I'm gonna use BSF with ffmpeg's `extract_extradata` so I'll let you know if I run into any issues.

Appreciated 😄

**Pull request merged**